src/hmem_cuda_gdrcopy.c

```diff
@@ -302,7 +302,7 @@ int cuda_gdrcopy_dev_register(struct fi_mr_attr *mr_attr, uint64_t *handle)
 	regbgn = (uintptr_t)mr_attr->mr_iov->iov_base;
 	regend = (uintptr_t)mr_attr->mr_iov->iov_base + mr_attr->mr_iov->iov_len;
 	regbgn = regbgn & GPU_PAGE_MASK;
-	regend = (regend & GPU_PAGE_MASK) + GPU_PAGE_SIZE;
+	regend = ((regend -1) & GPU_PAGE_MASK) + GPU_PAGE_SIZE;
 	reglen = regend - regbgn;
```

Comment:
now reglen is off by 1.

Reply:
They are not because (regend - 1) is aligned with GPU_PAGE_MASK, which is like to divide by GPU page size. 

It only matters if `regend` is exactly on GPU page boundary.  In other case `(regend -1) & GPU_PAGE_MASK` is same as `regend & GPU_PAGE_MASK`.

Reply:
You may be able to use the macros in ofi.h to help here.  See ofi_get_page_end(), ofi_get_page_start(), ofi_get_aligned_size()

Reply:
Yes, will do that

Reply:
Done.